public/modules/custom/helfi_ahjo/src/Services/AhjoService.php

```diff
@@ -0,0 +1,295 @@
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $guzzleClient;
+
+  /**
+   * Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * AHJO Service constructor.
+   *
+   * @param \Drupal\Core\Extension\ModuleExtensionList $extension_list_module
+   *   The module extension list.
+   * @param \Drupal\helfi_ahjo\Utils\TaxonomyUtils $taxonomyUtils
+   *   Taxonomy utils for tree.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \GuzzleHttp\ClientInterface $guzzleClient
+   *   A fully configured Guzzle client to pass to the dam client.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   Messenger service.
+   */
+  public function __construct(
+    ModuleExtensionList $extension_list_module,
+    TaxonomyUtils $taxonomyUtils,
+    EntityTypeManagerInterface $entity_type_manager,
+    ClientInterface $guzzleClient,
+    MessengerInterface $messenger
+  ) {
+    $this->moduleExtensionList = $extension_list_module;
+    $this->taxonomyUtils = $taxonomyUtils;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->guzzleClient = $guzzleClient;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('extension.list.module'),
+      $container->get('helfi_ahjo.taxonomy_utils'),
+      $container->get('entity_type.manager'),
+      $container->get('http_client'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getConfig(): ImmutableConfig {
+    return \Drupal::config('helfi_ahjo.config');
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function fetchDataFromRemote($orgId = 00001, $maxDepth = 9999): string {
+    $config = self::getConfig();
+    if (strlen($orgId) < 5) {
+      $orgId = sprintf('%05d', $orgId);
+    }
+
+    if (strlen($maxDepth) < 4) {
+      $maxDepth = sprintf('%04d', $maxDepth);
+    }
+
+    try {
+      $url = sprintf("%s/fi/ahjo-proxy/org-chart/$orgId/$maxDepth?api-key=%s", $config->get('base_url'), $config->get('api_key'));
+
+      $response = $this->guzzleClient->request('GET', $url);
+
+      return $response->getBody()->getContents();
+    }
+    catch (ClientException $e) {
+      throw new \Exception($e->getResponse()->getStatusCode());
+    }
+
+  }
+
+  /**
+   * @param array $childData
+   * @param array $operations
+   * @param int $parentId
+   *
+   * Recursive set all information from ahjo api.
+   *
+   * @return array|mixed
+   */
+  public function setAllBatchOperations($childData = [], &$operations = [], $parentId = 0) {
+    foreach ($childData as $content) {
+      $content['parentId'] = $parentId;
+      $operations[] = [$this->createTaxonomyTermsBatch(), [$content]];
```

Comment:
operations must have a callable name function.. here you are calling the method on the spot